src/main/java/hudson/plugins/gradle/injection/InitScriptTemplate.java

```diff
@@ -0,0 +1,41 @@
+package hudson.plugins.gradle.injection;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class InitScriptTemplate {
+
+    private static final Map<String, String> REPLACEMENTS;
+
+    static {
+        Map<String, String> replacements = new HashMap<>();
+        Arrays.stream(InitScriptVariables.values())
+            .filter(v -> !v.getTemplateName().isEmpty())
+            .forEach(v -> replacements.put(v.getTemplateName(), v.getTemplateTargetName()));
+        REPLACEMENTS = Collections.unmodifiableMap(replacements);
+    }
+
+    public static String replacePlaceholders(String template, Map<String, String> replacements) {
```

Comment:
🤔 Should we avoid this replacement and rename the env vars to remove the `JENKINSGRADLEPLUGIN` prefix?

Reply:
Would make things easier. I leave the decision up to you.

Reply:
Now I also feel like it would be better to just set the correct environment variables. In some plugins (e.g. in GitLab templates), it is a bit tricky to implement this replacement, so maybe script variables should just become our API for using the script.

Reply:
@welandaz what do you think?

Reply:
I'd also go with removing that prefix. I can't remember why it was introduced in the first place, but it seems unnecessary.